src/main/java/io/gravitee/policy/oauth2/Oauth2Policy.java

```diff
@@ -278,21 +333,33 @@ private Completable introspectAndValidateAccessToken(HttpExecutionContext ctx, S
      *      error="invalid_token",
      *      error_description="The access token expired"
      */
-    private Completable sendError(HttpExecutionContext ctx, String responseKey) {
-        String headerValue = BEARER_AUTHORIZATION_TYPE + " realm=\"gravitee.io\"";
+    private Completable sendError(BaseExecutionContext ctx, int errorStatus, String responseKey, String failureMessage) {
+        if (ctx instanceof HttpPlainExecutionContext httpPlainExecutionContext) {
+            String headerValue = BEARER_AUTHORIZATION_TYPE + " realm=\"gravitee.io\"";
 
-        ctx.response().headers().add(HttpHeaderNames.WWW_AUTHENTICATE, headerValue);
+            httpPlainExecutionContext.response().headers().add(HttpHeaderNames.WWW_AUTHENTICATE, headerValue);
 
-        return ctx.interruptWith(new ExecutionFailure(UNAUTHORIZED_401).key(responseKey).message(OAUTH2_UNAUTHORIZED_MESSAGE));
+            return httpPlainExecutionContext.interruptWith(new ExecutionFailure(errorStatus).key(responseKey).message(failureMessage));
+        }
+        // FIXME: Kafka Gateway - manage interruption with Kafka.
+        return Completable.error(new Exception(responseKey));
```

Comment:
Can you create a ticket to handle the interruption with Kafka and point to these fixme so we don't miss them.

Reply:
https://gravitee.atlassian.net/browse/APIM-7533